src/Console/CommandScanner.php

```diff
@@ -64,18 +64,11 @@ public function scanCore(): array
     protected function inflectCommandNames(array $commands): array
     {
         foreach ($commands as $i => $command) {
-            $command['name'] = str_replace('_', ' ', $command['name']);
-            $command['fullName'] = str_replace('_', ' ', $command['fullName']);
-            $commands[$i] = $command;
-
-            // Maintain backwards compatibility for schema_cache as it is likely
-            // hooked up to people's deploy tooling
-            if (strpos($command['name'], 'schemacache') === 0) {
-                $compat = [
-                    'name' => str_replace('schemacache', 'schema_cache', $command['name']),
-                    'fullName' => str_replace('schemacache', 'schema_cache', $command['fullName']),
-                ];
-                $commands[] = $compat + $command;
+            $name = $command['class']::getDefaultName();
```

Comment:
Would we also want this same behavior for app & plugin shells?

Reply:
yes , I'll add ,and I think this is Full BC

Reply:
It is if we include a default implementation.

Reply:
if user uses `UpdateTableCommand` this will show `update table` 
but documentation (in cake 3) says this should be `update_table`

Reply:
I can change the default to `underscore` if it should be